File: ServiceBook/ClassLibrary1/Models/WorkingPoint.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ServiceBook.DAL.Models
{
    public class WorkingPoint
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int ID { get; set; }
        public string Country { get; set; }

        public string City { get; set; }
        public int Flag { get; set; }
        public string Street { get; set; }
        
        public string Nr { get; set; }  
        public int VSCID { get; set; }
        [ForeignKey("VSCID")]
        public virtual VehicleServiceCompany VehicleServiceCompany { get; set; }
        public virtual ICollection<Employee> Employees { get; set; }

        public virtual ICollection<ServiceIntervention> ServiceInterventions { get; set; }
        public virtual ICollection<Service> Services { get; set; }

    }
}

File: ServiceBook/ServiceBook/Scripts/Service/Models/WorkingPoint.js
function WorkingPoint(data)
{
    this.id=null;
    this.country=ko.observable(null);
    this.city=ko.observable(null);
    this.street=ko.observable(null);
    this.nr=ko.observable(null);
    this.vscoid=null;
    this.employees=ko.observableArray(null);
    this.serviceInterventions = ko.observableArray(null);
    this.flag = null;
    this.vscid = null;
    this.services = ko.observableArray(null);
    this.serviceCompany = null;
    if (data != null)
    {
        this.id = data.ID;
        this.country(data.Country);
        this.city(data.City);
        this.street(data.Street);
        this.nr(data.Nr);
        this.vscid = data.VSCID;
        this.flag = data.Flag;
        var employees = _.map(data.Employees, function (employee, index) {
            return new Employee(employee);

        });
        this.employees(employees);
        var serviceInterventions=_.map(data.ServiceInterventions, function (serviceIntervention, index) {
            return new ServiceIntervention(serviceIntervention);

        });
        this.serviceInterventions(serviceInterventions);

        var services = _.map(data.Services, function (service, index) {
            return new Service(service);

        });
        this.services(services);
        this.serviceCompany = new VehicleServiceCompany(data.VehicleServiceCompany);
    }
}



